### ISSUE TITLE
Cannot add new node to swarm cluster using docker_swarm

### ISSUE THREAD

**Issue by kimdre** [ansible-collections/community.docker#876]
##### SUMMARY

I'm trying to add an additional worker node to my swarm cluster but the ansible playbook when triying to join the second worker node. 

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
- hosts: server_type_worker
  tasks:
    - name: add workers to swarm
      docker_swarm:
        state: join
        advertise_addr: "{{private_ipv4}}"
        remote_addrs: ["{{hostvars['manager-1'].private_ipv4}}"]
        join_token: "{{hostvars['manager-1'].swarm_init_result.swarm_facts.JoinTokens.Worker}}"
```

##### COLLECTION VERSION

```paste below
# /home/kim/.ansible/collections/ansible_collections
Collection       Version
---------------- -------
community.docker 3.10.3 

# /usr/lib/python3.12/site-packages/ansible_collections
Collection       Version
---------------- -------
community.docker 3.9.0  
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = /home/kim/Git/Magicline/magicline-server-deployment/ansible/ansible.cfg
DEFAULT_HOST_LIST(/home/kim/Git/Magicline/magicline-server-deployment/ansible/ansible.cfg) = ['/home/kim/Git/Magicline/magicline-server-deployment/ansible/inventory/prod.hcloud.yaml']
DEFAULT_REMOTE_USER(/home/kim/Git/Magicline/magicline-server-deployment/ansible/ansible.cfg) = root
EDITOR(env: EDITOR) = code --wait
HOST_KEY_CHECKING(/home/kim/Git/Magicline/magicline-server-deployment/ansible/ansible.cfg) = False
INVENTORY_ENABLED(/home/kim/Git/Magicline/magicline-server-deployment/ansible/ansible.cfg) = ['hcloud']
PAGER(env: PAGER) = most
```

##### OS / ENVIRONMENT

Fedora 40 and also Github Actions on Linux Runner
Worker Node has Ubuntu 24.04 LTS

##### STEPS TO REPRODUCE

```yaml
...
- hosts: server_type_worker
  tasks:
    - name: add workers to swarm
      docker_swarm:
        state: join
        advertise_addr: "{{private_ipv4}}"
        remote_addrs: ["{{hostvars['manager-1'].private_ipv4}}"]
        join_token: "{{hostvars['manager-1'].swarm_init_result.swarm_facts.JoinTokens.Worker}}"
```

##### ACTUAL RESULTS

worker-1 already exists, worker-2 is the new node.
```paste below
TASK [add workers to swarm] ********************************************************************************************************************************************************************************************************
ok: [worker-1]
fatal: [worker-2]: FAILED! => {"changed": false, "msg": "Error connecting: Error while fetching server API version: HTTPConnection.request() got an unexpected keyword argument 'chunked'"}
```

**Comment by felixfontein** [comment]
You probably want to upgrade Docker SDK for Python on the target where the module is executed.

This bug was tracked in https://github.com/docker/docker-py/issues/3113 and got fixed in Docker SDK for Python 7.0.0. You can alternatively downgrade urllib3 to a version < 2.

**Comment by kimdre** [comment]
Thank you! After installing docker on the new fresh node, the python docker SDK was still on 5.0.3 instead of >7.0.0 so I had to install the latest version manually. This fixed it for me. :)
